drivers/client_access_control/spec/models/grda_warehouse/hud/client_visibility_spec.rb

```diff
@@ -658,4 +658,121 @@
+        scope = window_source_client.enrollments_for_verified_homeless_history
+        expect(scope.count).to eq 1
+
+        scope = non_window_source_client.enrollments_for_verified_homeless_history
+        expect(scope.count).to eq 1
+      end
+    end
+    describe 'when visible_in_window config selected' do
+      before do
+        config.update(verified_homeless_history_method: :visible_in_window)
+      end
+      it 'enrollments visible in the window are included' do
+        scope = window_source_client.enrollments_for_verified_homeless_history
+        expect(scope.count).to eq 1
+
+        scope = non_window_source_client.enrollments_for_verified_homeless_history
+        expect(scope.count).to eq 0
+      end
+    end
+    describe 'when visible_to_user config selected' do
+      let!(:user) { create :user }
+      before do
+        config.update(verified_homeless_history_method: :visible_to_user)
+        user.roles << can_view_clients
+      end
+      it 'enrollments visible to user are included' do
+        # confirm client has 1 enrollment, but it's not included because it's not visible
+        expect(non_window_source_client.service_history_enrollments.count).to eq 1
+        expect(non_window_source_client.enrollments_for_verified_homeless_history(user: user).count).to eq 0
+
+        # add visibility and confirm it's included
+        user.add_viewable(non_window_project)
+        expect(non_window_source_client.enrollments_for_verified_homeless_history(user: user).count).to eq 1
+      end
+    end
+
+    describe 'when release config selected' do
+      let!(:user) { create :user }
+      before do
+        config.update(verified_homeless_history_method: :release)
+      end
+      describe 'and client has valid release in users CoC' do
+        before do
+          user.coc_codes = ['ZZ-999']
+          non_window_source_client.update(
+            housing_release_status: non_window_source_client.class.full_release_string,
+            consent_form_signed_on: 5.days.ago,
+            consent_expires_on: Date.current + 1.years,
+            consented_coc_codes: ['ZZ-999'],
+          )
+        end
+        it 'all enrollments included' do
+          expect(non_window_source_client.enrollments_for_verified_homeless_history(user: user).count).to eq 1
+        end
+      end
+
+      describe 'and client has valid release in any CoC, and user does not have assigned coc_codes' do
+        before do
+          user.coc_codes = []
+          non_window_source_client.update(
+            housing_release_status: non_window_source_client.class.full_release_string,
+            consent_form_signed_on: 5.days.ago,
+            consent_expires_on: Date.current + 1.years,
+            consented_coc_codes: ['ZZ-999'],
+          )
+        end
+        it 'all enrollments included' do
+          expect(non_window_source_client.enrollments_for_verified_homeless_history(user: user).count).to eq 1
+        end
+      end
```

Comment:
I'm not 100% sure if this is correct, can you confirm @eanders?

This is saying:

If the user has `coc_codes = []`, the `:release` config setting is set, and the client has a valid release in ANY coc, then ALL of their enrollments are included. In other words, `client.release_valid?(coc_codes: [])` returns true if the client has any consents.

Reply:
That doesn't sound right.  If the method is `:release`, and the user doesn't have any assigned `coc_codes`,  we should only show clients with a release in "ALL" CoCs.  That said, I _think_ in the situation where the method is `:release` a user should always have some `coc_codes`, so it's probably safe to show nothing if we get there and the user doesn't have any.

Reply:
Alright, that makes more sense. I figured it was the case that all users _should_ have CoC codes for installations where `:release` is configured. So, it doesn't seem worth it to create special logic to permit only releases that are designated for `All CoCs` in those cases. (I think that's what you're saying). 

Updated so that `release_valid?(coc_codes: [])` returns false: https://github.com/greenriver/hmis-warehouse/pull/2355/commits/730342acede666185a7ebb8f92df899ee56455d2